Add script to create release tarball.

bbannier opened the pull request:
This PR adds a script to create a release tarball and a GH action
attaching which creates and attaches the tarball to any release tags
pushed to the repo. We assume that that release tags are of the form
`v*`.

Closes #650.

rsmmr reviewed (dismissed):
Looks good. Let's add VERSION to `.gitignore` as well.

bbannier commented:
I added a GH action to upload the tarball created by the script. For that I changed the script to now always generate a file `spicy.tar.bz` which means we can hardcode its path in the action.

rsmmr commented:
> I added a GH action to upload the tarball created by the script. For that I changed the script to now always generate a file `spicy.tar.bz` which means we can hardcode its path in the action.

Can we still set a name when uploading? Would be nice if people downloading the tarball get a versioned file name.

bbannier commented:
> > I added a GH action to upload the tarball created by the script. For that I changed the script to now always generate a file `spicy.tar.bz` which means we can hardcode its path in the action.
> 
> Can we still set a name when uploading? Would be nice if people downloading the tarball get a versioned file name.

That's exactly what I did [here](https://github.com/zeek/spicy/blob/992873886ddd24ab3fdbefa889c1d6494bf95757/.github/workflows/create-release.yml#L38).

rsmmr commented:
> That's exactly what I did [here]

Excellent!

rsmmr commented on .github/workflows/create-release.yml:
```diff
@@ -0,0 +1,39 @@
+        uses: actions/checkout@v2
+      - name: Prepare archive
+        run: |
+          ./foo.sh
```
This correct?

bbannier replied:
Most definitely not, but a leftover from testing. Fixed now.
